src/Raven.Server/Documents/RetireAttachmentsSender.cs

```diff
@@ -186,6 +213,14 @@ internal async Task<int> RetireAttachments(int batchSize, long maxItemsToProcess
                                 continue;
                             }
                         }
+                        finally
+                        {
+                            // Wait for all direct uploaders to finish
+                            foreach (var uploader in directUploaders.Values)
+                            {
+                                uploader.Reset();
```

Comment:
`Reset` is doing a wait?

Reply:
Wait for all threads to finish & logs exceptions if needed:
https://github.com/ravendb/ravendb/blob/4119158f181d9a61c13c53d383e0a0f891aacbab/src/Raven.Server/Documents/PeriodicBackup/FileUploaderBase.cs#L83